pennylane/ops/pulse/convenience_functions.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This file contains convenience functions for pulse programming."""
+from typing import Callable, List, Tuple, Union
+
+has_jax = True
+try:
+    import jax.numpy as jnp
+except ImportError:
+    has_jax = False
+
+
+def constant(windows: List[Tuple[float]]):
+    """Returns a callable ``f(p, t)`` that returns ``p`` inside the time
+    windows defined in ``windows``.
+
+    Args:
+        windows (Tuple[float, Tuple[float]]): list of tuples containing time windows where x is
+            evaluated
+    """
+    return piecewise(x=lambda p, _: p, windows=windows)
+
+
+def piecewise(x: Union[float, Callable], windows: List[Tuple[float]]):
```

Comment:
I think it would make more sense if this function were called `window_function` or something similar. The difference between this and `pwc` would then be much clearer, and while this is technically a piecewise function, just calling it piecewise is very non-specific.  Maybe `rectangular_window` is better, since its more specific and accurate. Or `rectangular_envelope` could also be an option.

Reply:
But I think the `piecewise` and `pwc` naming has already been a source of confusion while organizing tasks, and I think it will continue to be confusing.

Reply:
Yep 💯 agree as this has been a point of confusion among ourselves already.
About the better naming I am not sure, I personally also find "rectangular" confusing as I would expect the output of that to be actually rectangular. I am also not sure what I would associate / expect of a "window" function 🤔 
Tagging @trbromley for UI input

Reply:
I think its fairly normal to describe a window that doesn't modify the shape of the input function as rectangular. This would contrast with f.x. a gaussian window that would also shape the amplitude of the input pulse within the window. 

In this case we multiply the input function by a rectangular function with an amplitude of 1, i.e. its not the output that is rectangular, but the other (fixed) input. The output is the two functions multiplied together. If we phrase it in this way in the docstring I think it would be fairly clear. But I'm certainly open to better suggestions, I just don't have one myself 😅

Reply:
Changed name to `rect`!